internal/core/data/event.go

```diff
@@ -192,6 +206,27 @@ func getEventById(id string) (contract.Event, error) {
 	return e, nil
 }
 
+// updateEventPushDateByChecksum updates the pushed dated for all events with a matching checksum which have not already been marked pushed
+func updateEventPushDateByChecksum(checksum string, ctx context.Context) error {
+	evts, err := dbClient.EventsByChecksum(checksum)
```

Comment:
To maintain a 1:1 event -> push activity, is it safe to assume earliest single event with the matching checksum is what caller intends to mark 'pushed'?

If so, we could order the events by e.createTime ascending and update just the earliest matched; i.e., index 0 as the one which was pushed.

Reply:
See comment below w/r/t expectation of checksum uniqueness. Right now the way the logic is written, all events for a given checksum will be marked pushed. If folks feel strongly, we can change but then there could possibly be orphans.

Reply:
@tsconn23 For the db.ErrNotFound to be meaningful (which I believe it could be) we should facilitate at least a 1:1 event received/pushed.

So I am only feeling strongly it should be one or the other: 
a) Mark all matching events as pushed (as it is) + do not produce error in cases where checksum is not found.  
b) Mark one matching event pushed (e.g., earliest in case time-sensitive but could be arbitrary) + produce error if/when checksum is not found.

Reply:
@tobiasmo1 Well we have to return a 404 if an invalid checksum is supplied (like "abcdef"). I don't see what that has to do with updating all matching records versus just one though. We're also talking about a slim edge case here.

Reply:
I suppose that when a device emits two events with same checksum... so long as we still push the 2nd event to subscribed apps (even if both were already marked pushed), it satisfies expected functionality.  I see a 404 as expected in this edge case, and it will occur 100% of the time when 2nd event gets marked as 'pushed'. . So if we update all matching records wouldn't the caller need to ignore such 404 errors instead of determining how/why the second event was 'lost'?

How about if we return 203 in response of first event being marked 'pushed' if/when there are > 1 matched events?

Reply:
> I see a 404 as expected in this edge case, and it will occur 100% of the time when 2nd event gets marked as 'pushed'

Not true. The query to fetch events by checksum looks like this
`mc.getEvents(bson.M{"checksum": checksum})`
And that's it. There's no criteria around whether it's been pushed or not, so those events will still be returned.

If you want a safeguard to ensure the Pushed property is only set if it's 0, then I can add that.

Reply:
Ah got it. Okay then we won't see a 404 and also handles case where caller retries to mark pushed for same event. Anyway, I am satisfied with current implementation so can resolve this thread.